### DISCUSSION TITLE
`ftd.type` improvement: list to font-family & new `color` property

### DISCUSSION THREAD

**Discussion by gargajit** [fastn-stack/fastn#1354] [Ideas & RFCs]
`ftd.type` can be improved by making the following changes to the `type` record's header properties:

- `font-family` should be made a list
- `color` property should be added

**type record:**
```ftd
-- record type:
optional ftd.font-size size:
optional ftd.font-size line-height:
optional ftd.font-size letter-spacing:
optional integer weight:
optional string font-family:
```

**Reasons for these suggestions:**

1. Making `font-family` a list will help the user to pass more than one font-family so that if the browser does not support the first `font-family` so it will check for the second `font-family` and so on, as `svelte` does:
![image](https://github.com/fastn-stack/fastn/assets/118595104/6b171cf4-5ae7-478a-a36a-4b0ee0bec633)

2. The `color` property will give the user the option to define a specific color as a requirement while creating the custom font type.

Currently, the user has to pass the `color` separately.
![image](https://github.com/fastn-stack/fastn/assets/118595104/bc28f583-a6f6-4faf-a9f9-46e07bd89578)

**Comment by amitu** [comment]
The font-family to list is good idea. Color is not. Color has to be undateable via color scheme packages.
